src/main/java/com/feedzai/commons/sql/abstraction/engine/DatabaseEngine.java

```diff
@@ -265,6 +266,23 @@
      */
     AbstractBatch createBatch(final int batchSize, final long batchTimeout, final String batchName);
 
+    /**
+     * Creates a new batch that periodically flushes a batch. A flush will also occur when the maximum number of
+     * statements in the batch is reached.
+     * <p/>
+     * Please be sure to call {@link AbstractBatch#destroy() } before closing the session with the database.
+     *
+     * @param batchSize     The batch size.
+     * @param batchTimeout  If inserts do not occur after the specified time, a flush will be performed.
+     * @param batchName     The batch name.
+     * @param batchListener Batch listener to execute custom behavior when the batch fails or succeeds to persist.
+     * @param logger        The logger.
+     * @return The batch.
+     */
+    default AbstractBatch createBatch(final int batchSize, final long batchTimeout, final String batchName, final BatchListener batchListener, final Logger logger) {
+        return createBatch(batchSize, batchTimeout, batchName, batchListener);
```

Comment:
this isn't passing the logger!?

Reply:
I wrote the method this way because it is necessary to have a default implementation to avoid breaking changes to existing code. But I see your point, and this implementation seems "wrong", since it ignores the last parameter. After thinking more about that I decided to change the default implementation to throw an exception. That way we still have the default implementation and prevent method calls without explicitly override. That is the same approach Java's `Iterator.remove()` default method implementation uses.